tests/ActivityApiTest.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+use Rogue\Models\Signup;
+use Rogue\Models\Event;
+use Faker\Generator;
+
+class ActivityApiTest extends TestCase
+{
+    /*
+     * Base URL for the Api.
+     */
+    protected $activityApiUrl = 'api/v2/activity';
+
+    /**
+     * Test for retrieving a user's activity with limit query param.
+     *
+     * GET /activity?limit=8
+     * @return void
+     */
+    public function testActivityIndexWithLimitQuery()
+    {
+        $this->json('GET', $this->activityApiUrl . '?limit=8');
+
+        $this->assertResponseStatus(200);
+
+        $this->seeJsonSubset([
+            'meta' => [
+                'pagination' => [
+                    'per_page' => 8
+                ],
+            ]
+        ]);
+    }
+
+    /**
+     * Test for retrieving a user's activity with page query param.
+     *
+     * GET /activity?page=3
+     * @return void
+     */
+    public function testActivityIndexWithPageQuery()
+    {
+        $this->json('GET', $this->activityApiUrl . '?page=3');
+
+        $this->assertResponseStatus(200);
+
+        $this->seeJsonSubset([
+            'meta' => [
+                'pagination' => [
+                    'current_page' => 3
+                ],
+            ]
+        ]);
+    }
+
+    /**
+     * Test for retrieving a user's activity with campaign_id query param.
+     *
+     * GET /activity?filter[campaign_id]=47
+     * @return void
+     */
+    public function testActivityIndexWithCampaignIdQuery()
+    {
+        $signup = factory(Signup::class)->create();
+        $event = Event::all()->first();
+        $event->signup_id = $signup->id;
+        $event->save();
+
```

Comment:
@DFurnes updated and got it to work this way. I couldn't use something like this to work:

```
$users = factory(App\User::class, 3)
           ->create()
           ->each(function ($u) {
                $u->posts()->save(factory(App\Post::class)->make());
            });
```

because in the Signup Factory, if I don't have 
```        
        'event_id' => function() {
            return factory(Event::class)->create()->id;
        },
```
it gets mad but then I think we are creating two events which is incorrect too? I'm not sure if the original way reads cleaner and is safer to create a signup instead of `Event::all()first->();` line?